paymentsheet/src/main/java/com/stripe/android/paymentsheet/specifications/Spec.kt

```diff
@@ -21,24 +20,33 @@ data class LayoutSpec(val elements: List<FormElementSpec>) {
         get() = elements.filterIsInstance<SectionSpec>().map { it.field }
 }
 
+data class IdentifierSpec(val value: String)
+
 /**
  * This is used to define each section in the visual form layout
  */
 
 sealed class FormElementSpec {
-    data class SectionSpec(val field: SectionFieldSpec) : FormElementSpec() {
-        sealed class SectionFieldSpec(val identifier: String) {
-            object Name : SectionFieldSpec("name")
+    data class SectionSpec(
+        val identifier: IdentifierSpec,
+        val field: SectionFieldSpec
+    ) : FormElementSpec() {
 
-            object Email : SectionFieldSpec("email")
+        sealed class SectionFieldSpec(val identifier: IdentifierSpec) {
```

Comment:
does `SectionFieldSpec` have to be nested within? seems it's not a subclass of any spec right? can we extract it out like `IdentifierSpec`?

Reply:
It doesn't, I can pull it out.